src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -2338,6 +2338,67 @@ public static Vector256<sbyte> Shuffle(Vector256<sbyte> vector, Vector256<sbyte>
             return result;
         }
 
+        /// <summary>Creates a new vector by selecting values from an input vector using a set of indices.
+        /// Behavior is platform-dependent for out-of-range indices.</summary>
+        /// <param name="vector">The input vector from which values are selected.</param>
+        /// <param name="indices">The per-element indices used to select a value from <paramref name="vector" />.</param>
+        /// <returns>A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.</returns>
+        /// <remarks>Unlike Shuffle, this method delegates to the underlying hardware intrinsic without ensuring that <paramref name="indices"/> are normalized to [0, 31].
+        /// On hardware with <see cref="Avx2"/> support, but not <see cref="Avx512Vbmi.VL"/>, indices are treated as modulo 32, and if the high bit is set, the result will be set to 0 for that element.
+        /// On hardware with <see cref="Avx512Vbmi.VL"/> support, indices are treated as modulo 32.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Avx2))]
+        [CompExactlyDependsOn(typeof(Avx512Vbmi.VL))]
+        public static Vector256<byte> ShuffleUnsafe(Vector256<byte> vector, Vector256<byte> indices)
+        {
+            if (Avx512Vbmi.VL.IsSupported)
+            {
+                return Avx512Vbmi.VL.PermuteVar32x8(vector, indices);
+            }
+
+            if (Avx2.IsSupported)
+            {
+                // check for constant value of indices where we can optimise further
+                if (RuntimeHelpers.IsKnownConstant(indices))
+                {
+                    // check for all in correct lane (or will result in 0)
```

Comment:
It's not clear to me what this means / what conditions we're special-casing here.

Should any of them be moved down to `Shuffle` or are we relying on the specific instructions applying an effective MOD 16?

Reply:
What I'm checking here is whether we have something like: `Shuffle(..., V256.Create(15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16))` where the first 16 are within 0-15, and the next 16 are within 16-31. The second case is similar but checks 0-15 for all. In these cases, and we could do for a few more, we can emit some more optimised code due to lack of needing to calculate both shuffles.

Reply:
I don't think Shuffle currently generates good code with only AVX2 (is my understanding anyway).
Would be good to fix that too, but I haven't done any JIT coding yet, so I didn't start there at least 😆

> Should any of them be moved down to `Shuffle` or are we relying on the specific instructions applying an effective MOD 16?

The intel docs specify how they work, so I don't think we need to be worried about it for this particular reason.